pkg/monitoring/profiler/http-client-profiler.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package profiler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"k8s.io/client-go/rest"
+)
+
+type counter struct {
+	counter     map[string]int
+	lock        sync.Mutex
+	isProfiling bool
+}
+
+var globalCounter counter
+
+func StartHttpProfiler() {
+	globalCounter.lock.Lock()
+	defer globalCounter.lock.Unlock()
+	globalCounter.counter = make(map[string]int)
```

Comment:
I am wondering if the Prometheus metrics `rest_client_request_latency_seconds_count` and `apiserver_request_total` show similar results.

If not, which different http calls are expected to be gather with this tool that is not exposed to Prometheus.
Also, if some http call metrics are not exposed to Prometheus, we should consider to expose them.